File: src/app/Models/MaterialTest.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use App\Traits\Auditable;

class MaterialTest extends Model
{
    use HasFactory, SoftDeletes, Auditable;

    protected $fillable = [
        'raw_material_entry_id',
        'test_date',
        'resistance_ohm_km',
        'elongation_pct',
        'check_winding',
        'check_cleanliness',
        'check_packaging',
        'check_identification',
        'result',
        'conducted_by',
        'approved_by'
    ];

    protected $casts = [
        'test_date' => 'date',
        'resistance_ohm_km' => 'float',
        'elongation_pct' => 'float',
        'check_winding' => 'boolean',
        'check_cleanliness' => 'boolean',
        'check_packaging' => 'boolean',
        'check_identification' => 'boolean',
    ];

    public function entry()
    {
        return $this->belongsTo(RawMaterialEntry::class, 'raw_material_entry_id');
    }
}

File: backend-dotnet/src/Ikhana.Domain/Entities/MaterialTest.cs
using Ikhana.Domain.Enums;

namespace Ikhana.Domain.Entities;

public class MaterialTest : BaseEntity, ISoftDelete, IAuditable
{
    public long RawMaterialEntryId { get; set; }

    public DateOnly TestDate { get; set; }
    public decimal ResistanceOhmKm { get; set; }
    public decimal? ElongationPct { get; set; }

    public bool? CheckWinding { get; set; }
    public bool? CheckCleanliness { get; set; }
    public bool? CheckPackaging { get; set; }
    public bool? CheckIdentification { get; set; }

    public TestResult Result { get; set; }

    public string? ConductedBy { get; set; }
    public string? ApprovedBy { get; set; }

    public DateTime? DeletedAt { get; set; }

    public RawMaterialEntry Entry { get; set; } = null!;
}
